Readme and license (#15)
Cambridge-ICCS/summer-school-scientific-vis: README.md

<img src="https://iccs.cam.ac.uk/sites/iccs.cam.ac.uk/files/logo2_1.png"  width="355" align="left">

<br><br><br><br><br>

# ICCS Scientific visualisation

![GitHub](https://img.shields.io/github/license/Cambridge-ICCS/summer-school-scientific-vis)
[![CC BY-NC-SA 4.0][cc-by-nc-sa-shield]][cc-by-nc-sa]

This repository contains documentation, resources, and code for the Scientific Visualisation with matplotlib session designed and delivered by [Jack Atkinson](https://jackatkinson.net/) ([**@jatkinson1000**](https://github.com/jatkinson1000))
and James Emberton ([**@j-emberton**](https://github.com/j-emberton)) of [ICCS](https://github.com/Cambridge-ICCS).  
This material was originally prepared for the [ICCS](https://iccs.cam.ac.uk/events/iccs-summer-school-2024) 2024 summer school.  

> All materials, including slides and videos, are available such that individuals can cover the course in their own time.

Reviewer: We have no videos, at least not yet.
```suggestion
All materials, including slides and workbooks, are available such that individuals can cover the course in their own time.
```

## Contents

- [Learning Objectives](#learning-objectives)
- [Teaching material](#teaching-material)
- [Preparation and prerequisites](#preparation-and-prerequisites)
- [Installation and setup](#installation-and-setup)
- [License information](#license)
- [Contribution Guidelines and Support](#contribution-guidelines-and-support)

# Learning Objectives

The key learning objective from this workshop is to build skills in scientific visualisation through python and its libraries.

However, more specifically we aim to:

> summarise key considerations when visualising scientific data,
> provide an understanding of the Matplotlib package and how to use it effectively,
> introduce the Cartopy package and show how it extends Matplotlib functionality for geospatial plotting,
> introduce common climate data formats and show how these can be parsed to generate plottable data,
> and encourage good research software engineering (RSE) practice while using code to generate scientific visualisations.

Reviewer: ```suggestion
- summarise key considerations when visualising scientific data,
- provide an understanding of the Matplotlib package and how to use it effectively,
- introduce the Cartopy package and show how it extends Matplotlib functionality for geospatial plotting,
- introduce common climate data formats and show how these can be parsed to generate plottable data, and
- encourage good research software engineering (RSE) practice while using code to generate scientific visualisations.
```

- A text editor - e.g. vim/[neovim](https://neovim.io/), [gedit](https://gedit.en.softonic.com/), [vscode](https://code.visualstudio.com/), [sublimetext](https://www.sublimetext.com/) etc. to open and edit code files
- A terminal emulator - e.g. [GNOME Terminal](https://help.gnome.org/users/gnome-terminal/stable/), [wezterm](https://wezfurlong.org/wezterm/index.html), [Windows Terminal (windows only)](https://learn.microsoft.com/en-us/windows/terminal/), [iTerm (mac only)](https://iterm2.com/)
- python virtual environment (see [Installation and setup](#installation-and-setup))

Note for Windows users: _We have linked suitable applications for windows in the above lists.
However, you may wish to refer to [Windows' getting-started with python information](https://learn.microsoft.com/en-us/windows/python/beginners)
for a complete guide to getting set up on a Windows system._

If you require assistance or further information with any of these please reach out to
us before a training session.

## Building

The slides are written using quarto.
To build them run:
```
quarto render Scientific_Vis.qmd 
```

> from this directory and inspect the output .html file.

Reviewer: ```suggestion
from the `presentation/` directory and inspect the output .html file (assuming you have [Quarto](https://quarto.org/) installed on your machine ).
```

## Installation and setup

There are three options for participating in this workshop for which instructions are provided below:

* via a [local install](#local-install)
* on [Google Colab](#google-colab)

> * on [binder](#binder)

Reviewer: I don't think these are set up yet...?
Happy to leave them in, but we should open an issue to make sure we enable them before the summer school.

Reviewer: Hi @j-emberton I think this one must have slipped through the cracks.

However, if you experience issues with the installation process or are unfamiliar with
the terminal/installation process there is the option to run the notebooks in
[Google Colab](#google-colab) or on [binder](#binder).

### Local Install

#### 1. Clone or fork the repository
Navigate to the location you want to install this repository on your system and clone
via https by running:
```
git clone https://github.com/Cambridge-ICCS/summer-school-scientific-vis.git
```
This will create a directory `summer-school-scientific-vis/` with the contents of this repository.

Please note that if you have a GitHub account and want to preserve any work you do
we suggest you first [fork the repository](https://github.com/Cambridge-ICCS/summer-school-scientific-vis/fork) 
and then clone your fork.
This will allow you to push your changes and progress from the workshop back up to your
fork for future reference.

#### 2. Create a virtual environment
Before installing any Python packages it is important to first create a Python virtual environment.
This provides an insulated environment inside which we can install Python packages 

> without polluting the operating systems' Python environment.

Reviewer: ```suggestion
without polluting the operating system's Python environment.
```
